### ISSUE TITLE
Introduce cookie settings

### ISSUE THREAD

**Issue by JulianDietzel** [bastislack/highline-freestyle#326]
> [!NOTE]
> Important follow up issue to #279. The following was moved to this separate issue to a) contain the scope of #279 and b) since this was not originally part of the issue. This should be addressed only once #279 is implemented.

Since in #279 video embeds are introduced we need a way to allow the user to consent to or decline the use of third party cookies. To do this, the following needs to be put in place:
- [x] A page (/part of a page) where the cookie settings can be set globally for the entire app. Adding a settings page is probably the most appropriate solution for this. This page should contain:
   - [x] A short description what these settings do (they allow us to show the videos but can give the embed provider the possibility to track the user. We ourselves don't track the user and don't place any cookies)
   - [x] A toggle for YouTube
   - [x] A toggle for Instagram
- [x] A system for storing the users preferences (some kind of in-browser storage)
- [x] A component which is shown instead of an embed if the user has not (yet) consented to cookies from the provider of the embed. This component should include:
   - [x]  A short description what this setting does (it allows us to show the video but can give the embed provider the possibility to track the user)
   - [x] The option to allow embeds from this provider permanently (/globally)
- [x] A short wiki entry for how this system works (where are the preferences stored)

**Comment by WaldemarLehner** [comment]
I have made something similar for my website. We could reuse that logic if you want :) 

https://waldemarlehner.de/en/privacy/

**Comment by JulianDietzel** [comment]
Started working on this. Current progress can be found on [feat/cookie-settings](https://github.com/bastislack/highline-freestyle/tree/feat/settings-page).

**Comment by JulianDietzel** [comment]
Only some translations of the `video.en.json` file into Spanish and French, and a short wiki entry are still missing. Then this will be done.

**Comment by JulianDietzel** [comment]
Implemented in #338
